resources/tests/org.eclipse.core.tests.resources/src/org/eclipse/core/tests/internal/resources/WorkspacePreferencesTest.java

```diff
@@ -33,103 +34,126 @@
 import org.eclipse.core.runtime.CoreException;
 import org.eclipse.core.runtime.IPath;
 import org.eclipse.core.runtime.Preferences;
-import org.eclipse.core.tests.resources.WorkspaceSessionTest;
-import org.eclipse.core.tests.session.WorkspaceSessionTestSuite;
+import org.eclipse.core.tests.harness.session.SessionTestExtension;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.junit.jupiter.api.io.TempDir;
+
+@SuppressWarnings("deprecation")
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+public class WorkspacePreferencesTest {
+
+	@TempDir
+	static Path tempDirectory;
+
+	@RegisterExtension
+	static SessionTestExtension sessionTestExtension = SessionTestExtension.forPlugin(PI_RESOURCES_TESTS)
+			.withWorkspaceAt(tempDirectory).create();
 
-public class WorkspacePreferencesTest extends WorkspaceSessionTest {
 	private IWorkspace workspace;
 	private Preferences preferences;
 
-	@Override
-	protected void setUp() throws Exception {
-		super.setUp();
+	@BeforeEach
+	public void retrieveWorkspaceAndPreferences() throws CoreException {
 		workspace = ResourcesPlugin.getWorkspace();
 		preferences = ResourcesPlugin.getPlugin().getPluginPreferences();
-		workspace.setDescription(Workspace.defaultWorkspaceDescription());
 	}
 
-	private void setDefaultWorkspaceDescription() throws CoreException {
+	@AfterEach
+	public void restoreWorkspaceDescription() throws CoreException {
 		workspace.setDescription(Workspace.defaultWorkspaceDescription());
 	}
 
 	/**
 	 * Tests properties state in a brand new workspace (must match defaults).
 	 */
+	@Test
+	@Order(1)
```

Comment:
Why is the order of the tests relevant now?

Reply:
It has also been relevant before. The `WorkspaceSessionTestSuite` (that was used before) sorts the tests by default (based on a lexicographic order of test method names). This does not happen with the JUnit 5 `SessionTestExtension` anymore (as also documented in its Javadoc), which is both an active decision (as it improves comprehensibility when making the order of test cases explicit if relevant) and technically easier to realize with JUnit 5 extensions.

Reply:
It just occurred to me: if `WorkspaceSessionTestSuite` sorted the tests in lexicographical order then shouldn't you annotate the class with `@TestMethodOrder(MethodOrderer.Alphanumeric.class)`? That way new tests would follow the same order they would have followed in the old `WorkspaceSessionTestSuite`